### PR TITLE
Update dwc2_stm32.h

### LINKED ISSUE #1431: Support of STM32H7xx with only 1 USB
### Operating System

Windows 10

### Board

NUCLEO-H723ZG

### Firmware

Any firmware

### What happened ?

The STM32H7 family have 2 sub families:
- H74x and H75x have 2 USB, and USB2 have an internal FS PHY. No problem with this sub family.
- H72x, H73x, H7Ax, H7Bx have only 1 USB named USB1 with an internal FS PHY (this USB1 is equivalent to the USB2 of the other sub family). And in the CMSIS files of this sub family the values USB_OTG_FS_PERIPH_BASE and OTG_FS_IRQn are not declared, so the file synopsys/dwc2/dwc2_stm32.h produce compile errors.

This can be solved by adding theses lines to the file dwc2_stm32.h after line 58:
[https://github.com/hathach/tinyusb/blob/master/src/portable/synopsys/dwc2/dwc2_stm32.h#L58](https://github.com/hathach/tinyusb/blob/master/src/portable/synopsys/dwc2/dwc2_stm32.h#L58)

  #if (! defined USB2_OTG_FS)
    // H7 with only 1 USB port: H72x / H73x / H7Ax / H7Bx
    // USB_OTG_FS_PERIPH_BASE and OTG_FS_IRQn not defined in CMSIS files
    #define USB_OTG_FS_PERIPH_BASE  USB1_OTG_HS_PERIPH_BASE
    #define OTG_FS_IRQn             OTG_HS_IRQn
  #endif

Successfully tested on NUCLEO-H723ZG (1 USB) and NUCLEO-H743ZI (2 USB).

### How to reproduce ?

Configure a project for one of this 2nd sub family of the H7.

### Debug Log as txt file

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Nikitarc**
Support STM32H7 with only 1 USB port: H72x / H73x / H7Ax / H7Bx
Avoid GCC compiler warnings when -pedantic option is used (strict ISO C)

**Describe the PR**
The STM32H7 family have 2 sub families:

- H74x and H75x have 2 USB, and USB2 have an internal FS PHY. No problem with this sub family.
- H72x, H73x, H7Ax, H7Bx have only 1 USB named USB1 with an internal FS PHY (this USB1 is equivalent to the USB2 of the other sub family). And in the CMSIS files of this sub family the values USB_OTG_FS_PERIPH_BASE and OTG_FS_IRQn are not declared, so the file synopsys/dwc2/dwc2_stm32.h produce compile errors.

Add to tusb_options.h
`typedef int make_iso_compilers_happy ;`

**Additional context**
Successfully tested on NUCLEO-H723ZG (1 USB) and NUCLEO-H743ZI (2 USB).
fixes #1431

### CODE REVIEW

File: src/portable/synopsys/dwc2/dwc2_stm32.h [hunk 1]
```diff
@@ -56,6 +56,11 @@
   #define EP_FIFO_SIZE_FS 4096
   #define EP_MAX_HS       9
   #define EP_FIFO_SIZE_HS 4096
+  #if (! defined USB2_OTG_FS)
+    // H7 with only 1 USB port: H72x / H73x / H7Ax / H7Bx
+    // USB_OTG_FS_PERIPH_BASE and OTG_FS_IRQn not defined
+    #define USB_OTG_FS_PERIPH_BASE  USB1_OTG_HS_PERIPH_BASE
+    #define OTG_FS_IRQn             OTG_HS_IRQn
```
Comment (hathach) [inline src/portable/synopsys/dwc2/dwc2_stm32.h:63]: seems like you miss an #endif, please make sure you compile the example and verify the changes works

**Review by hathach — CHANGES_REQUESTED** [review]
missing #endif

File: src/portable/synopsys/dwc2/dwc2_stm32.h (hunk 1 above)
Comment (Nikitarc) [reply src/portable/synopsys/dwc2/dwc2_stm32.h:63]: Sorry this is my first PR, and I got confused.
I'm sure it works now.

File: src/tusb_option.h [hunk 2]
```diff
@@ -27,6 +27,9 @@
 #ifndef _TUSB_OPTION_H_
 #define _TUSB_OPTION_H_
 
+// To avoid GCC compiler warnings when -pedantic option is used (strict ISO C)
+typedef int make_iso_compilers_happy ;
```
Comment (perigoso) [inline src/tusb_option.h:31]: why is this necessary? what was the error?

Comment (Nikitarc) [reply src/tusb_option.h:31]: This is not an error correction.
Only good programming practice: avoid displaying dozens of warning messages when compiling these modules.
If a warning appears, you have to go see what it is, and maybe an important warning is buried in it.

Comment (perigoso) [reply src/tusb_option.h:31]: sorry, I meant what warning this was correcting

**Comment by Nikitarc** [comment]
These warning:

..\TinyUSB\src\class\msc\msc_host.c:499: warning: ISO C forbids an empty translation unit [-Wpedantic]
